app/representers/proxy_representer.rb

```diff
@@ -47,10 +47,6 @@ class ProxyRepresenter < ThreeScale::Representer
   property :jwt_claim_with_client_id, if: ->(*) { oidc? }
   property :jwt_claim_with_client_id_type, if: ->(*) { oidc? }
 
-  def policies_config
-    represented.policies_config.to_json
```

Comment:
Would changing this to `represented.policies_config.as_json` work for both - JSON and XML?

Reply:
Imho that was the issue, we returned the JSON but we need to keep it compatible with older versions where we return an array of hashes.

Reply:
Also I think `spec/acceptance/api/proxy_spec.rb` needs to be amended to catch this issue properly. A quick run of the test file with `#as_json` still has it passed but it also passed with `#to_json` so I don't know whether the end result is precisely what we expect or not :)

Reply:
`#as_json` should return an array of maps. That's why I suggest it - to work with both representers. An oversight on my side to use `#to_json`.

Reply:
```
PRY-SHELL
◈────────────────────────────────────────────────────────────────────────────────◈

Frame number: 0/80

From: /home/avalon/ws/repos/porta/app/representers/proxy_representer.rb:52 ProxyRepresenter#policies_config:

    50: def policies_config
    51:   binding.pry_shell
 => 52:   represented.policies_config.as_json
    53: end

[1] pry(#<ProxyRepresenter::JSON>)> represented.policies_config.as_json
=> [{"name"=>"apicast",
  "version"=>"builtin",
  "configuration"=>{},
  "enabled"=>true}]
[2] pry(#<ProxyRepresenter::JSON>)> 
```

Reply:
Another good place to add a test for this might be `ProxiesControllerTest#test_show`. On neither place the returned data format is checked.